joomla/joomla-cms
# [4.0] Feature: Adding User(s) Model + View to front-end

Pull Request to **add** a **new Model** + **View** for **Users/User** to the **front-end**.

Note: this PR is an improved version of https://github.com/joomla/joomla-cms/pull/24799
It has been decoupled from Articles (the previous version showed (the number) of articles that a User had created).

### Summary of Changes
The front-end does not have a model + view to display a list of Users.
You cannot use the back-end model of com_users in the front-end because that model has some authentication so that you cannot get data from the Users table if you are not logged in.

This PR adds a Front-end View + Model for Users and User.
The view can be added to the Front-end by creating a new menu item type "Users > List Users".
The Users View supports Custom Fields.

### Testing Instructions
Create some Users that you assign to some User Group (e.g. Author and Editor).

In **Menus** > Main Menu, **create a New Menu Item** of type "Users > **List Users**"

![joomla-users1](https://user-images.githubusercontent.com/1217850/58484938-5b98d980-8163-11e9-8c74-5498184b7087.png)

Under the "**Users View**" tab, configure what the new menu item should display:
- User Groups: select a few
- Select if you want to see the Page Header or not
- Configure if you want to see the Last Visit date of that User
- Configure the Date Format if you want to show it differently
- Configure if you want to see the Online status of that User

![joomla-users2](https://user-images.githubusercontent.com/1217850/58484937-5b98d980-8163-11e9-9f4a-9ab1643a54c2.png)

### Expected result
**The Front-end** should have a new View that **displays a list of Users**. 
You should be able to navigate the list.

![joomla-users3](https://user-images.githubusercontent.com/1217850/58484935-5b98d980-8163-11e9-8e5e-d1b85f576c8a.png)

If you select a User, you should see the **User page** with currently only the name of the user:

![joomla-users4](https://user-images.githubusercontent.com/1217850/58484934-5b98d980-8163-11e9-8648-1d486fa3347f.png)

The User View supports Custom Fields that have been added to the User.
For the moment you can display all other Fields (besides the Name it already displays) + Custom Fields them by creating a Template Override.
